### ISSUE TITLE
When manually switching the connection it resets the search path

### ISSUE THREAD

**Issue by rpbaltazar** [rails-on-services/apartment#27]
## Steps to reproduce

```
Apartment::Tenant.switch!('existing_tenant')
puts Apartment::Tenant.current # prints 'existing_tenant'
ActiveRecord::Base.connected_to(role: :writing) do
  puts Apartment::Tenant.current # prints 'existing_tenant'
  puts Apartment.connection.schema_search_path # does not print 'existing_tenant'
end
```

## Expected behavior

```
Apartment::Tenant.switch!('existing_tenant')
puts Apartment::Tenant.current # prints 'existing_tenant'
ActiveRecord::Base.connected_to(role: :writing) do
  puts Apartment::Tenant.current # prints 'existing_tenant'
  puts Apartment.connection.schema_search_path # prints 'existing_tenant'
end
```

## System configuration

* Database: Postgresql 12.1

* Apartment version: 2.3.0

* Apartment config (in `config/initializers/apartment.rb` or so):

  * `use_schemas`: `true`

* Rails (or ActiveRecord) version: 6.0.2.1

* Ruby version: 2.6.5

**Comment by rpbaltazar** [comment]
It also happens when switching the connection on a model level:

```
Apartment::Tenant.switch('existing_tenant')
User::Account.connected_to(role: :reading) do
  User::Account.first(10)
end
```

will return the first 10 user accounts under the default schema instead of the switched schema

**Comment by rpbaltazar** [comment]
Resolved with #28
